src/lib/__tests__/InstantSearch-test.tsx

```diff
@@ -1047,7 +1047,8 @@ describe('dispose', () => {
 
     search.start();
 
-    await runAllMicroTasks();
+    await Promise.resolve();
+    await Promise.resolve();
```

Comment:
Calling it once didn't pass the test?

Reply:
yes, there's two micro-tasks that need to happen, this implementation makes that clearer than wait(0) IMO

Reply:
oh I see, but I feel like this is testing too much detail here because what we want to know is if `onRender` is called after rendering happens. We do not want to test if onRender is called two micro-tasks after start().

So I'm wondering what if we 

* `const runAllMicroTasks = () => wait(0)` and use it normally
* use `await Promise.resolve()` when we need to test specific timing issue?